File: sub-web-service/src/main/java/org/github/web/module/sys/menu/service/IMenuService.java
package org.github.web.module.sys.menu.service;

import lombok.NonNull;
import org.github.mysql.web.base.entity.SysMenuMbpEntity;
import org.github.spring.restful.json.JSONArrayReturn;
import org.github.spring.restful.json.JSONDataReturn;
import org.github.spring.restful.json.JSONPageReturn;
import org.github.web.module.sys.menu.model.bo.QueryMenuListBO;
import org.github.web.module.sys.menu.model.vo.QueryMenuListVO;

import java.util.List;

public interface IMenuService {
  @NonNull JSONDataReturn<Boolean> delMenuList(@NonNull List<Long> ids);

  @NonNull JSONDataReturn<SysMenuMbpEntity> queryMenuInfo(@NonNull Long menuId);

  @NonNull JSONPageReturn<QueryMenuListVO> queryMenuPage(@NonNull QueryMenuListBO bo);

  @NonNull JSONArrayReturn<SysMenuMbpEntity> queryMenuTree();

  @NonNull JSONDataReturn<Boolean> saveOrUpdate(@NonNull SysMenuMbpEntity bo);
}

File: sub-web-service/src/main/java/org/github/web/module/sys/menu/service/IMenuService.kt
package org.github.web.module.sys.menu.service

import org.github.spring.restful.json.JSONDataReturn
import org.github.mysql.mydb.base.entity.SysMenuEntity
import org.github.web.module.sys.menu.model.bo.QueryMenuListBO
import org.github.spring.restful.json.JSONPageReturn
import org.github.web.module.sys.menu.model.vo.QueryMenuListVO
import org.github.spring.restful.json.JSONArrayReturn

interface IMenuService {
  fun delMenuList(ids: List<Long>): JSONDataReturn<Boolean>

  fun queryMenuInfo(menuId: Long): JSONDataReturn<SysMenuEntity?>

  fun queryMenuPage(bo: QueryMenuListBO): JSONPageReturn<QueryMenuListVO>

  fun queryMenuTree(): JSONArrayReturn<SysMenuEntity>

  fun saveOrUpdate(bo: SysMenuEntity): JSONDataReturn<Boolean>
}
